must-gather/collection-scripts/gather

```diff
@@ -52,43 +55,56 @@ cluster_resources+=(clusterrolebindings)
 cluster_resources+=(persistentvolumes)
 cluster_resources+=(clusterversion)
 cluster_resources+=(machineconfigpool)
+cluster_resources+=(uiplugin)
 cluster_resources+=(customresourcedefinitions)
 
-log "- BEGIN inspecting cluster resources..." | tee "${LOGFILE_PATH}"
+log "- BEGIN inspecting cluster resources and namespaces..." | tee -a "${LOGFILE_PATH}"
+
 for cr in "${cluster_resources[@]}" ; do
-  log "-- BEGIN inspecting cluster resource ${cr} ..." >> "${LOGFILE_PATH}"
+  log "-- BEGIN inspecting cluster resource ${cr} ..." | tee -a "${LOGFILE_PATH}"
   oc adm inspect --cache-dir=${KUBECACHEDIR} --dest-dir="${BASE_COLLECTION_PATH}" "${cr}"  >> "${LOGFILE_PATH}" 2>&1 &
   pids+=($!)
 done
-log "- END inspecting cluster resources..." >> "${LOGFILE_PATH}"
+
+for ns in "${namespace_resources[@]}" ; do
+  log "-- BEGIN inspecting namespace ${ns} ..." | tee -a "${LOGFILE_PATH}"
+  oc adm inspect --cache-dir=${KUBECACHEDIR} --dest-dir="${BASE_COLLECTION_PATH}" "ns/${ns}"  >> "${LOGFILE_PATH}" 2>&1 &
+  pids+=($!)
+done
+log "- END inspecting cluster resources..." | tee -a "${LOGFILE_PATH}"
 
 # namespace-scoped resources
 resources="pods,roles,rolebindings,configmaps,serviceaccounts,events,installplans,subscriptions,clusterserviceversions,logfilemetricexporter"
```

Comment:
I think one reason for the complexity is that `oc adm inspect` will not work when specifying non-existing CRDs.

Wouldn't it be simpler to build this "namespaced resource types" list dynamically based on what CRDs are available (LokiStack, ElasticSearch, Kibana, ClusterLogForwarder, ClusterLogging, ...) and then just do a `oc adm inspect` call for each of the namespaces that we think is relevant?

The same approach could be used for the cluster-scoped resources. There I think the only "optional" resource type is the UIPlugin.

Reply:
@xperimental , AFAIK, if the CRDs doesn't exist, the `oc adm inspect [resource]` fails with the same error than `oc get [resource]`. But they should be there if the operator is installed (I don't know if the operator reconcile the CRDs if they are manually deleted by mistake while the operator is still installed.
It could be simpler to make a list of available CRDs (to be honest, I don't know), but in any case, the `oc adm inspect` command should be used instead of `oc get`. And that will require to re-do the full must-gather scripts.

Reply:
@xperimental , one option could be using `|| true` at the end of the command, and redirect the output from the `oc adm inspect` to the log (in case there are errors like the resource not found).
That way, it will not be needed to check all the CRDs (like for example if Loki, the UIPlugin, etc are not installed in the cluster), so the script could be less complex. But for doing that, it will be required to refactor the whole script. For now, it would be good to have the missing information collected as this PR tries.